iroh-bytes/src/baomap.rs

```diff
@@ -0,0 +1,288 @@
+    /// Progress when computing the outboard
+    ///
+    /// There will be multiple of these messages for an id
+    OutboardProgress { id: u64, offset: u64 },
+    /// Done computing the outboard
+    ///
+    /// This comes after `Size` and zero or more `OutboardProgress` messages
+    OutboardDone { id: u64, hash: Hash },
+}
+
+/// The import mode describes how files will be imported.
+///
+/// This is a hint to the import trait method. For some implementations, this
+/// does not make any sense. E.g. an in memory implementation will always have
+/// to copy the file into memory. Also, a disk based implementation might choose
+/// to copy small files even if the mode is `Reference`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum ImportMode {
+    /// This mode will copy the file into the database before hashing.
+    ///
+    /// This is the safe default because the file can not be accidentally modified
+    /// after it has been imported.
+    #[default]
+    Copy,
+    /// This mode will reference the file in place and assume it is unchanged after import.
+    ///
+    /// This has a large performance and storage benefit, but it is less safe since
+    /// the file might be modified after it has been imported.
+    Reference,
+}
+/// The import mode describes how files will be imported.
+///
+/// This is a hint to the import trait method. For some implementations, this
+/// does not make any sense. E.g. an in memory implementation will always have
+/// to copy the file into memory. Also, a disk based implementation might choose
+/// to copy small files even if the mode is `Reference`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum ExportMode {
+    /// This mode will copy the file to the target directory.
+    ///
+    /// This is the safe default because the file can not be accidentally modified
+    /// after it has been exported.
+    #[default]
+    Copy,
+    /// This mode will move the file to the target directory and then reference it from
+    /// the database.
+    ///
+    /// This has a large performance and storage benefit, but it is less safe since
+    /// the file might be modified in the target directory after it has been exported.
+    Reference,
+}
+
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub enum ExportProgress {
+    /// Starting to export to a file
+    ///
+    /// This will be the first message for an id
+    Start {
+        id: u64,
+        hash: Hash,
+        path: PathBuf,
+        stable: bool,
+    },
+    /// Progress when copying the file to the target
+    ///
+    /// This will be omitted if the store can move the file or use copy on write
+    ///
+    /// There will be multiple of these messages for an id
+    Progress { id: u64, offset: u64 },
+    /// Done exporting
+    Done { id: u64 },
+}
+
+/// Progress updates for the provide operation
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ValidateProgress {
+    /// started validating
+    Starting {
```

Comment:
maybe `Start` as well to keep consistent with `ExportProgress`?

Reply:
Yeah, this one is much older than the others.

You can spend a lot of time bikeshedding whether this should be `Start` (imperative) or `Starting` (which for me seems like a state name and not a good name for a state transition event). But I guess `Start` it is because it is short...

Reply:
I don't mind either, as long as it is consistent